### PR TITLE
Allow pilot to start w/o kubernetes cluster

### PR DESCRIPTION
**Pull request by Nino-K**
**What this PR does / why we need it**:

- Allows pilot to start with inmemory config store
- This work is required to introduce CloudFoundry adapter

[#151465998]

Signed-off-by: Nino Kodabande <nkodabande@suse.com>

This PR allows pilot to start without kubernetes, this is done through allowing pilot to start with an in-memory config store. This work is a first step to introduce Cloudfoundry platform adapter.

**Release note**:
```release-note
Allows pilot to start with in-memory configStore. 
```

### CODE REVIEW

File: pilot/cmd/pilot-discovery/main.go [hunk 1]
```diff
@@ -213,22 +240,26 @@ var (
...
-			flags.admissionArgs.ServiceNamespace = flags.namespace
-			flags.admissionArgs.DomainSuffix = flags.controllerOptions.DomainSuffix
-			flags.admissionArgs.ValidateNamespaces = []string{
-				flags.controllerOptions.WatchedNamespace,
-				flags.namespace,
-			}
-			admissionController, err := admit.NewController(client, flags.admissionArgs)
-			if err != nil {
-				return fmt.Errorf("failed to create validation admission controller: %v", err)
+			if flags.configStore == kubernetesConfigStore && registered[platform.KubernetesRegistry] {
```
Comment (ayj) [inline pilot/cmd/pilot-discovery/main.go]: AdmissionController creation should only be predicated on the configStore. It's possible to use k8s ConfigStore with Consul/Eureka service registry in which case we'd still want to create the admission controller.

**Comment by Nino-K** [comment]
The CLA issue with this PR should now be fixed.

Thanks,
@flawedmatrix & @Nino-k

File: pilot/cmd/pilot-discovery/main.go (hunk 1 above)
Comment (Nino-K) [reply pilot/cmd/pilot-discovery/main.go]: @ayj Updated as per conversation [here](https://github.com/istio/old_pilot_repo/issues/1566#issuecomment-340972929).

**Comment by nmittler** [comment]
@Nino-K I just took a quick look at this PR, sorry for coming late to the party.  I suspect that #2003 is a superset of what is done here ... or am I missing something?  If you'd like something additional, I'm sure we can work it into #2003. WDYT?

**Comment by rshriram** [comment]
@Nino-K I suggest you take a look a @nmittler 's PR. Its a bit more comprehensive. The only difference is the inMemoryConfigStore you are creating. Nathan's PR has a file based version of that.

**Comment by flawedmatrix** [comment]
@nmittler, @rshriram I agree, after looking at the work done in #2003, I believe it obsoletes the work we've done in this PR. We'd love to just use that PR and just rebase our addition of the Cloudfoundry registry on top of that. What can we do to speed the process along?

**Comment by nmittler** [comment]
@flawedmatrix I think we're just trying to sort out some of the failures that @rshriram mentioned in https://github.com/istio/istio/pull/2019#issuecomment-350197726.  I'm new to the project and it's unclear to me if those failures are a result of this PR ... any help here would be much appreciated.

**Comment by Nino-K** [comment]
@rshriram Looking at @nmittler  [PR](https://github.com/istio/istio/pull/2003) seems to align with what we were originally attempted to do. We will go ahead and close this PR and base the new one off #2003 

Thanks
@aaronshurley & @Nino-K

**Pull request closed without merging**